File: Week -2/Qs11.py
def factorial(n):
    """Calculate the factorial of a number."""
    if n == 0 or n == 1:
        return 1
    fact = 1
    for i in range(2, n + 1):
        fact *= i
    return fact

def combination(n, r):
    """Calculate the binomial coefficient nCr."""
    return factorial(n) // (factorial(r) * factorial(n - r))

def print_pascals_triangle(rows):
    """Print Pascal's Triangle with the given number of rows."""
    for i in range(rows):
        # Print leading spaces
        print(' ' * (rows - i - 1), end='')
        # Print numbers in the row
        for j in range(i + 1):
            print(combination(i, j), end=' ')
        print()

# Taking input for the number of rows
rows = int(input("Enter the number of rows for Pascal's Triangle: "))

# Printing Pascal's Triangle
print_pascals_triangle(rows)

File: Week -2/Qs11.c
#include <stdio.h>

// Function to calculate factorial of a number
long factorial(int n) {
    long fact = 1;
    for (int i = 1; i <= n; i++) {
        fact *= i;
    }
    return fact;
}

// Function to calculate nCr (combination)
long combination(int n, int r) {
    return factorial(n) / (factorial(r) * factorial(n - r));
}

// Function to print Pascal's Triangle
void printPascalsTriangle(int rows) {
    for (int i = 0; i < rows; i++) {
        // Print leading spaces
        for (int j = 0; j < rows - i - 1; j++) {
            printf(" ");
        }
        // Print numbers in the row
        for (int j = 0; j <= i; j++) {
            printf("%ld ", combination(i, j));
        }
        printf("\n");
    }
}

int main() {
    int rows;

    // Taking input for the number of rows
    printf("Enter the number of rows for Pascal's Triangle: ");
    scanf("%d", &rows);

    // Printing Pascal's Triangle
    printPascalsTriangle(rows);

    return 0;
}
